debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/OracleConnection.java

```diff
@@ -353,39 +353,39 @@ public Long getSessionStatisticByName(String name) throws SQLException {
     /**
      * Returns whether the given table exists or not.
      *
-     * @param tableName table name, should not be {@code null}
+     * @param tableId table id, should not be {@code null}
      * @return true if the table exists, false if it does not
      * @throws SQLException if a database exception occurred
      */
-    public boolean isTableExists(String tableName) throws SQLException {
-        return queryAndMap("SELECT COUNT(1) FROM USER_TABLES WHERE TABLE_NAME = '" + tableName + "'",
-                rs -> rs.next() && rs.getLong(1) > 0);
-    }
-
     public boolean isTableExists(TableId tableId) throws SQLException {
+        if (Strings.isNullOrBlank(tableId.schema())) {
+            return queryAndMap("SELECT COUNT(1) FROM USER_TABLES WHERE TABLE_NAME = '" + tableId.table() + "'",
+                    rs -> rs.next() && rs.getLong(1) > 0);
+        }
         return queryAndMap("SELECT COUNT(1) FROM ALL_TABLES WHERE OWNER = '" + tableId.schema() + "' AND TABLE_NAME = '" + tableId.table() + "'",
                 rs -> rs.next() && rs.getLong(1) > 0);
     }
 
     /**
      * Returns whether the given table is empty or not.
      *
-     * @param tableName table name, should not be {@code null}
+     * @param tableId table id, should not be {@code null}
      * @return true if the table has no records, false otherwise
      * @throws SQLException if a database exception occurred
      */
-    public boolean isTableEmpty(String tableName) throws SQLException {
-        return getRowCount(tableName) == 0L;
+    public boolean isTableEmpty(TableId tableId) throws SQLException {
+        return getRowCount(tableId) == 0L;
     }
 
     /**
      * Returns the number of rows in a given table.
      *
-     * @param tableName table name, should not be {@code null}
+     * @param tableId table id, should not be {@code null}
      * @return the number of rows
      * @throws SQLException if a database exception occurred
      */
-    public long getRowCount(String tableName) throws SQLException {
+    public long getRowCount(TableId tableId) throws SQLException {
+        final String tableName = new TableId(null, tableId.schema(), tableId.table()).toDoubleQuotedString();
```

Comment:
Is it necessary to create a new `TableId`? I'd argu that even the `tableName` vriable is not necessary.